src/relations/database_provides.py

```diff
@@ -180,3 +183,28 @@ def reconcile_users(
             if relation not in requested_users:
                 relation.delete_user(shell=shell)
         logger.debug(f"Reconciled users {event=}, {router_endpoint=}")
+
+    def get_status(self, event) -> typing.Optional[ops.StatusBase]:
+        """Report non-active status."""
+        requested_users = []
+        exceptions: list[status_exception.StatusException] = []
+        for relation in self._interface.relations:
+            try:
+                requested_users.append(
+                    _RequestedUser(relation=relation, interface=self._interface, event=event)
+                )
+            except _RelationBreaking:
+                pass
+            except (remote_databag.IncompleteDatabag, _UnsupportedExtraUserRole) as exception:
+                exceptions.append(exception)
+        # Always report unsupported extra user role
+        for exception in exceptions:
+            if isinstance(exception, _UnsupportedExtraUserRole):
+                return exception.status
+        if requested_users:
+            # At least one relation is active—do not report about inactive relations
+            return
+        for exception in exceptions:
+            if isinstance(exception, remote_databag.IncompleteDatabag):
```

Comment:
it might be a good idea to add a suffix `Exception` so that we know that `remote_databag.IncompleteDatabagException` is an exception (without having to trace back to the definition of the class to see that it's a subclass of an exception)

same applies to all subclasses of `StatusException`